cmd/clusterctl/client/cluster/cert_manager_test.go

```diff
@@ -21,11 +21,84 @@ import (
 	"time"
 
 	. "github.com/onsi/gomega"
+	admissionregistration "k8s.io/api/admissionregistration/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/util/wait"
 	"sigs.k8s.io/cluster-api/cmd/clusterctl/client/config"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/internal/scheme"
 	"sigs.k8s.io/cluster-api/cmd/clusterctl/internal/test"
 )
 
+func Test_certManagerClient_getManifestObjects(t *testing.T) {
+
+	tests := []struct {
+		name      string
+		expectErr bool
+		assert    func(*testing.T, []unstructured.Unstructured)
+	}{
+		{
+			name:      "it should not contain the cert-manager-leaderelection ClusterRoleBinding",
+			expectErr: false,
+			assert: func(t *testing.T, objs []unstructured.Unstructured) {
+				for _, o := range objs {
+					if o.GetKind() == "ClusterRoleBinding" && o.GetName() == "cert-manager-leaderelection" {
+						t.Error("should not find cert-manager-leaderelection ClusterRoleBinding")
+					}
+				}
+			},
+		},
+		{
+			name:      "the MutatingWebhookConfiguration should have sideEffects set to None ",
```

Comment:
Does this apply to the validation web hook as well?

Reply:
ohhh.... I did not check the yaml but I saw

> This is also consistent with the ValidatingWebhookConfiguration for cert-manager-webhook.

Reply:
There is a ValidationWebhookConfiguration however it already had the `sideEffects: None` in the original v0.11.0 cert-manager.yaml.

Reply:
I added these tests to track only the changes being made to this file in order to avoid any possible regressions if we ever decide to update cert-manager.yaml.

Reply:
If `sideEffects` can give problems IMO we should add a test that ensures this field isn't removed from the ValidationWebhookConfiguration (In order to prevent a regression in the yaml).
But I want block on this, so I leave to you the final choice

Reply:
@fabriziopandini I can add that test to ensure that it's there as well. I'm never opposed to adding more tests 😆

Reply:
I added a test for the ValidatingWebhookConfiguration to ensure that `sideEffects: None`.